### PR TITLE
fix(pagination): adds doc on reset behavior, changes the name of setter

### PR DESCRIPTION
**Pull request by bobylito**
The `helper.setCurrentPage` is deprecated in favor of `helper.setPage`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -651,6 +651,30 @@ helper.getCurrentPage();
...
+that any operation that may change the number of page will reset the pagination to page 0.
+
+This may lead to some unexpected behavior. For example:
+
+```js
+// helper.state.page = 0
+helper.setPage(4);
+// helper.state.page = 4
+helper.setQuery('foo');
+// helper.state.page = 0
```
Comment (vvo) [inline README.md]: I would not talk about helper.state, just:

```js
helper.setPage(4);
helper.setQuery('foo');
helper.search(); // page will be 0, even if 4 was asked
```

Comment (bobylito) [inline README.md]: Fair enough!

File: README.md [hunk 2]
```diff
@@ -651,6 +651,30 @@ helper.getCurrentPage();
...
+```
+
+Non exhaustive list of operations that trigger a reset :
```
Comment (vvo) [inline README.md]: Non exhaustive because there are two many of them? Maybe we can do the opposite list (or it's not needed?)

Comment (vvo) [inline README.md]: `reset :` no space before ` :`

File: README.md (hunk 1 above)
Comment (vvo) [inline README.md]: And maybe say that you need to do the setPage last if you really really want page 4

File: src/algoliasearch.helper.js [hunk 3]
```diff
@@ -366,22 +366,33 @@ AlgoliaSearchHelper.prototype.nextPage = function() {
  * @return {AlgoliaSearchHelper}
  */
 AlgoliaSearchHelper.prototype.previousPage = function() {
-  return this.setCurrentPage(this.state.page - 1);
+  return this.setPage(this.state.page - 1);
 };
 
+function setCurrentPage(page) {
+  if (page < 0) throw new Error('Page requested below 0.');
```
Comment (vvo) [inline src/algoliasearch.helper.js]: Do we need to test this?

File: README.md (hunk 1 above)
Comment (bobylito) [inline README.md]: What about using getPage (well getCurrentPage for now :P, will fix that)?

Comment (vvo) [inline README.md]: In the example? I would say we only need to explain the trap people will fall into but I understand the concern about also explaining that the internal state is modified (for power users)

File: src/algoliasearch.helper.js (hunk 3 above)
Comment (bobylito) [inline src/algoliasearch.helper.js]: Well that's the current behavior and the idea behind when implemented was "best do it before than later"

**Comment by bobylito** [comment]
I think we're good here. Thanks for the feedback @vvo

**Pull request merged**